### PR TITLE
Bug: Emit doesn't update 2 way bound props

### PR DESCRIPTION
**Pull request by nandi95**

### CODE REVIEW

File: tests/props.spec.ts [hunk 1]
```diff
@@ -82,4 +83,32 @@ describe('props', () => {
       object: {}
     })
   })
+
+  it('should return the updated props on 2 way binding', async () => {
```
Comment (nandi95) [inline tests/props.spec.ts:87]: This file is probbly the wrong place for this test, please advise where it would be best.

File: src/emit.ts [hunk 2]
```diff
@@ -48,7 +48,7 @@ function recordEvent(
   events: Events,
   vm: ComponentInternalInstance,
   event: string,
-  args: Events[number]
+  args: unknown[]
```
Comment (nandi95) [inline src/emit.ts:51]: This I typed wrong in the previous PR, fixed here

File: src/emit.ts [hunk 3]
```diff
@@ -64,4 +64,16 @@ function recordEvent(
 
   // Record the event message sent by the emit
   events[cid][event].push(args)
+
+  if (event.startsWith('update:')) {
+    if (args.length !== 1) {
+      throw new Error(
+        'Two-way bound properties have to emit a single value.' +
```
Comment (nandi95) [inline src/emit.ts:71]:
```suggestion
        'Two-way bound properties have to emit a single value. ' +
```

**Review by lmiller1990 — APPROVED** [review]
I am somewhat surprised this works. We had some discussion around v-model testing, I think this is pretty good.

https://github.com/vuejs/vue-test-utils-next/discussions/279

File: tests/props.spec.ts (hunk 1 above)
Comment (lmiller1990) [reply tests/props.spec.ts:87]: Here is probably fine. Probably could re-think the test layout at some point.

**Comment by lmiller1990** [comment]
We may revisit and revert this @nandi95. Based on https://github.com/vuejs/vue-test-utils-next/issues/440#issuecomment-790538633

In that case, you'd need to just change to use a `vModel` mounting option.

**Pull request merged**